lib/ex_aws/ec2.ex

```diff
@@ -2,1924 +2,2795 @@ defmodule ExAws.EC2 do
   @moduledoc """
   Operations on AWS EC2
 
-  ## Basic Operations
-
   A selection of the most common operations from the EC2 API are implemented here.
   http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_Operations.html
 
-  ### Filters
-
-  Many of the `Describe` endpoints allow you to filter based on a number of attributes.
-  Refer to the AWS documentation for the specified method for the acceptable filter values.
 
-  When supplying atoms, underscores will be converted to a dash for compatibility
-  with the API parameters.
-
-  ### Examples
+  Examples of how to use this:
   ```elixir
-  ExAws.EC2.create_vpc("10.0.0.0/16")
-  ExAws.EC2.describe_instances(filters: [image_id: "ami-1ecae776"])
-  ExAws.EC2.describe_instances(filters: ["network-interface.availability-zone": "us-east-1a"])
-  ExAws.EC2.describe_instances(filters: ["tag:elasticbeanstalk:environment-name": "demo"])
-  ExAws.EC2.describe_instance_status(instance_id: ["i-e5974f4c"])
+  alias ExAws.EC2
+
+  EC2.terminate_instances(["i-123456", "i-987654"], [dry_run: true])
+  EC2.register_image("Test", [block_device_mappings: [
+    [device_name: "/dev/sda1", ebs: [snapshot_id: "snap-1234567890abcdef0"]],
+    [device_name: "/dev/sdb",  ebs: [snapshot_id: "snap-1234567890abcdef1"]],
+    [device_name: "/dev/sdc",  ebs: [volume_size: 100]]
+  ]])
   ```
   """
+  use ExAws.Utils,
+  format_type: :xml,
+  non_standard_keys: %{
+    assign_ipv6_address_on_creation: "AssignIpv6AddressOnCreation.Value",
+    enable_dns_support: "EnableDnsSupport.Value",
+    enable_dns_hostnames: "EnableDnsHostnames.Value",
+    map_public_ip_on_launch: "MapPublicIpOnLaunch.Value"
+  }
 
-  import ExAws.Utils
-
-  @version "2015-10-01"
-
-  ########################
-  ### Instance Actions ###
-  ########################
+  @version "2016-11-15"
 
-  @type instance_types :: [
```

Comment:
We shouldn't keep track of the instance_types or any of the type specs that AWS only accepts. This becomes tedious and most likely will not keep up with AWS' changes.